### ISSUE TITLE
Issue updating SelectEntry in onChange method for filtered list

### ISSUE THREAD

**Issue by Ragin-LundF** [fyne-io/fyne#1118]
### Describe the bug:
I want to filter the drop-down when the user is typing something into the `SelectEntry` text field.
For that, I've created code like described in the `example code` section.

If I debug the code, I can see, that the elements on `optionSelectEntry` are correct, but the UI isn't refreshing. Also an `optionSelectEntry.Refresh()` has no effect.

### To Reproduce:
see example code

### Example code:
**widget code**
```go
optionSelectEntry := widget.NewSelectEntry(filteredListForSelect(nil))
optionSelectEntry.PlaceHolder = "Type or select an option"
optionSelectEntry.OnChanged = func(input string) {
	optionSelectEntry.SetOptions(filteredListForSelect(&input))
}
```

**example for `filterListForSelect` function**
```go
func filteredListForSelect(filter *string) (resultList []string) {
	allOptionsList := []string{"Option A", "Selection B", "Method C", "Button D"}
	if filter == nil || *filter == "" {
		return allOptionsList
	} else {
		for _, option := range allOptionsList {
			if strings.Contains(option, *filter) {
				resultList = append(resultList, option)
			}
		}
		return resultList
	}
}
```

### Device (please complete the following information):
 - OS: found on MacOS
 - Version: 10.15.5
 - Go version: 1.14.4
 - Fyne version: 1.3.0

**Comment by andydotxyz** [comment]
If you're able to test the patch in #1119 that would be helpful - I think it fixes the issue, but let me know :)

**Comment by andydotxyz** [comment]
Merged for v1.3.1
